test/e2e/scripts/generate-parameters.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+##### A script to generate a unique namespace #####
+##### and a parameters file for a workflow    #####
+
+
+##### Exit on error #####
+set -e
+
+##### Functions #####
+
+usage()
+{
+    echo '    Usage: ./generate_parameters.sh [[-w workflow -g workflow_group] | [-h]]
+    Example: ./generate_parameters.sh -g workflow_group -w workflow
+    Flags:
+    -w, --workflow         workflow name
+    -g, --workflow-group   workflow group name
+    -o, --output-file      generated yaml file name (default parameters.yaml)
+    -d, --workflows-dir    the directory where workflows are defined (default ../argo-workflows)'
+}
+
+check_yq_installed()
+{
+    if ! [ -x "$(command -v yq)" ]; then
+        echo 'Error: yq is not installed.'
+        exit 1
+    fi
+}
+
+validate_input()
+{
+    # Validate workflow name characters
+    if ! [[ $WORKFLOW =~ ^[0-9a-zA-Z._-]+$ ]]; then
```

Comment:
Allowing `.` and `_` will potentially lead to invalid namespace formats (per k8s constraints [here](https://github.com/kubernetes/apimachinery/blob/ed135c5b96450fd24e5e981c708114fbbd950697/pkg/api/validation/generic.go#L62)).

Reply:
good catch 👍